packages/playwright-core/src/server/bidi/bidiBrowser.ts

```diff
@@ -473,6 +445,39 @@ function toBidiSameSite(sameSite: channels.SetNetworkCookie['sameSite']): bidi.N
   return bidi.Network.SameSite.None;
 }
 
+function getProxyConfiguration(proxySettings?: types.ProxySettings): bidi.Session.ManualProxyConfiguration | undefined {
+  if (!proxySettings)
+    return undefined;
+
+  const proxy: bidi.Session.ManualProxyConfiguration = {
+    proxyType: 'manual',
+  };
+  const url = new URL(proxySettings.server);  // Validate proxy server.
+  switch (url.protocol) {
+    case 'http:':
+      proxy.httpProxy = url.host;
+      break;
+    case 'https:':
+      proxy.httpsProxy = url.host;
```

Comment:
I think this one should be rather `sslProxy` (see https://www.w3.org/TR/webdriver-bidi/#cddl-type-sessionmanualproxyconfiguration)
```suggestion
      proxy.sslProxy = url.host;
```